NBitcoin.Altcoins/ForkIdTransaction.cs

```diff
@@ -47,7 +47,7 @@ public override uint256 GetSignatureHash(Script scriptCode, int nIn, SigHash nHa
 			if(UsesForkId(nHashType))
 				nForkHashType |= ForkId << 8;
 
-			if((SupportSegwit && sigversion == HashVersion.Witness) || UsesForkId(nHashType))
+			if((SupportSegwit && sigversion == HashVersion.Witness) || (UsesForkId(nHashType) && ForkId != 42))
```

Comment:
Please do not change this code. I have no way to know that it did not break bgold or bcash.

Reply:
How would you prefer I skip over this conditional for BTCP?

Reply:
I would override `GetSignatureHash` with a class deriving from `ForkIdTransaction`, say BPrivateTransaction.

Reply:
Or Making UseForkId virtual, and only overriding this in BPrivateTransaction